### ISSUE TITLE
Converting "post" scope to "create update" scope fails if scope is not an array

### ISSUE THREAD

**Issue by Dreamseer** [indieweb/wordpress-indieauth#120]
When logging in at [OwnYourGram](https://ownyourgram.com) using WP IndieAuth I got this warning:

`Warning: array_search() expects parameter 2 to be array, string given in /…/wp-content/plugins/indieauth/includes/class-indieauth-authorization-endpoint.php on line 230`

The authorization fails, thus I cannot login.

Seems like an `is_array( $scope )` check around [this change](https://github.com/indieweb/wordpress-indieauth/commit/70e89dc82697c36dcf77a25100f53fbbc8aeabe8) does the trick:

```php
if ( is_array( $scope ) ) {
	$search = array_search( 'post', $scope, true );
	if ( is_numeric( $search ) ) {
		unset( $scope[ $search ] );
		$scope = array_unique( array_merge( $scope, array( 'create', 'update' ) ) );
	}
}
```

**Comment by dshanske** [comment]
Will issue a fix. Thanks for the feedback

**Comment by dshanske** [comment]
Should be resolved via #121. Advise if issue.
